fix: render cells that has no content but styles

kxxt opened the pull request:
Fix #181

I am not confident about this fix since I only briefly looked at the code base. So this PR should be reviewed carefully to avoid breaking other things.

kxxt committed:
fix: render cells that has no content but styles

Fix #181

a-kenji commented:
Thank you for taking the time, creating this pr!
And sorry for the ci problems :).

On the surface this does make sense. 
I will try it out for a bit to check for regressions.

kxxt commented:
> On the surface this does make sense. I will try it out for a bit to check for regressions.

I think the test failure is indeed caused by this PR but I don't know if my PR is wrong or the test snapshot is wrong.

a-kenji commented:
I believe the snapshot needs to be updated.

The `vttest` `script` snapshots usually test very constrained functionality.
But I also added few snapshots that test a little more interaction, to check if general behavior changes.
So the test failing here is not a bad thing.

kxxt commented:
> I believe the snapshot needs to be updated.
> 
> The `vttest` `script` snapshots usually test very constrained functionality. But I also added few snapshots that test a little more interaction, to check if general behavior changes. So the test failing here is not a bad thing.

Thanks for your explanation, I have updated the failing snapshot.

a-kenji commented on src/vt100_imp.rs:
```diff
@@ -37,8 +37,11 @@ impl Cell for vt100::Cell {
+    if screen_cell.has_contents() {
+        buf_cell.set_symbol(&screen_cell.contents());
+    } else {
+        buf_cell.set_symbol(" ");
```
Did you find this is needed? 

I removed  `buf_cell.set_symbol(" ");` and saw no noticeable difference.

kxxt replied:
> Did you find this is needed?
> 
> I removed `buf_cell.set_symbol(" ");` and saw no noticeable difference.

Fixed. On a second thought I think it's not needed.

a-kenji reviewed (approved):
LGMT, thank you for the contribution :+1: .

kxxt commented:
Hi, @a-kenji. Could you please publish a new release(or prerelease) for this fix? Thanks!

a-kenji commented:
@kxxt,
Not sure if you got pinged on the release, but this fix is now in `v0.11.0`,

kxxt commented:
> @kxxt, Not sure if you got pinged on the release, but this fix is now in `v0.11.0`,

Thanks a lot for your new release! I didn't get pinged but several days ago I found the bug magically disappeared, then I figured out that `cargo update` command I ran updated tui-term to 0.1.11: https://github.com/kxxt/tracexec/commit/52a26c42304527abf0d3f08882ba01175f0592e0

BTW there are some other issues regarding input with htop inside tui-term, which I am going to try to fix when I got more free time: https://github.com/kxxt/tracexec/issues/11
